File: misc/compression-code-recognizes-loaded-files.rb
files = ['foo', 'bar', 'baz']
j_code = 'tinymce.each("' + files.join(',') + '".split(","),function(f){tinymce.ScriptLoader.markDone(tinyMCE.baseURL+"/"+f+".js");});';

puts j_code

File: misc/compression-code-recognizes-loaded-files.php
<?php
$files = array('foo', 'bar', 'baz');
$buffer = '';
$buffer .= 'tinymce.each("'
    . implode(',', $files)
    . '".split(","),function(f){tinymce.ScriptLoader.markDone(tinyMCE.baseURL+"/"+f+".js");});';

print $buffer . "\n";